lib/OptimizationManopt/src/OptimizationManopt.jl

```diff
@@ -361,23 +269,27 @@ function build_gradF(f::OptimizationFunction{true})
         f.grad(G, θ)
         return riemannian_gradient(M, θ, G)
     end
+    return g
 end
 
 function build_hessF(f::OptimizationFunction{true})
     function h(M::AbstractManifold, H1, θ, X)
         H = zeros(eltype(θ), length(θ))
+        # an Optimization function has both hess (the matrix) and hv (Hessian with direction)
+        # we need hv here
```

Comment:
```suggestion
```

Reply:
`Hessian(M, p, X)` is always the hv thing here. 

The simple reason is the one thing that we had to fix when using DiffEq as well: Vector dimension is often not the same as manifold dimension.

So the hessians in the JuliaManifolds system are always what is called `hv` here.
You would need a Basis of the tangent spaces (domain and range) to build a matrix representation. And that is usually too costly. So I dan see how someone confused that.

I am not sure where the last comment was from, that is not fromme, I meant the one you deleted here reminding of the difference between Hess and hv – since gain, the Hess form does not exist in JuliaManifolds and its Hess is here hv.